src/apps/app_search.erl

```diff
@@ -41,6 +41,22 @@ add_entry(undefined, _, _, _) -> do_nothing;
 add_entry(PID, Name, Value, ID) ->
 	PID ! {add_tx, Name, Value, ID}.
 
+get_tags_by_id(TXID) ->
+	get_tags_by_id(whereis(http_search_node), TXID).
+get_tags_by_id(PID, TXID) ->
+	PID ! {get_tags, TXID, self()},
+	receive Tags ->
```

Comment:
Receiving anything and assuming it's what you expect is dangerous. Wrap the message and receive it like this instead:
`receive {tags, Tags} ->`